packages/core/src/dispatcher.ts

```diff
@@ -167,16 +167,21 @@ export class Dispatcher {
     return this._shutdownSignal
       .abortable((signal) => this._ipfs.dag.put(record, { signal: signal }))
       .then((cid) => {
-        this.dagNodeCache.set(cid.toString(), record)
+        this.ipldCache.setRecord(cid, record)
         return cid
       })
   }
 
   async getIpfsBlock(cid: CID): Promise<Uint8Array> {
-    return await this._shutdownSignal.abortable((signal) => {
-      // @ts-ignore
-      return this._ipfs.block.get(cid, { signal, offline: !this.enableSync })
-    })
+    const found = this.ipldCache.get(cid)
+    if (found) {
+      return found.block
```

Comment:
Can we add metrics here (think this is the right spot)?

Reply:
Added.